src/vocabularyChecker/detectWrongSyntax.test.ts

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import {collectAllInvalidWords, collectInvalidWordsInIssues, getAllIssues} from "./detectWrongSyntax";
+import {getInvalidWords, logMultipleOccurences, logMultipleWords, logSingleWord} from "./detectWrongSyntax";
+import {Position} from "./detectWrongSyntax";
+
+describe("getInvalidWords", () => {
+	it("Returns the correct word for a normal text", () => {
+		const wrongWords = getInvalidWords("fu bar bloedsinn"); // fu and bar are indeed included in allowedVocab
+		expect(wrongWords).toEqual(["bloedsinn"]);
+	});
+
+	it("Returns an empty array, if input is the empty string", () => {
+		const wrongWords = getInvalidWords("");
+		expect(wrongWords).toEqual([]);
+	});
+
+	it("Returns an invalid word only once, if it occurs multiple times", () => {
+		const wrongWords = getInvalidWords("bloedsinn bloedsinn");
+		expect(wrongWords).toEqual(["bloedsinn"]);
+	});
+
+// maybe it should not care about case sensitivity, but that's more of a design choice
+	it("is case sensitive", () => {
+		const wrongWords = getInvalidWords("fu Fu");
+		expect(wrongWords).toEqual(["Fu"]);
+	});
+
+	it("Does not return a wrong word, if a string consists only of a whitespace", () => {
+		const wrongWords = getInvalidWords(" ");
+		expect(wrongWords).toEqual([]);
+	});
+
+	it("Does not detect a wrong word, if there is none", () => {
+		const wrongWords = getInvalidWords("fu bar");
+		expect(wrongWords).toEqual([]);
+	});
+
+	it("Works for long strings", () => {
+		const wrongWords = getInvalidWords(`This is just a really long String to test some extreme case so
+		the next letters will only be some copies of the letter a.
+		aaaaaaaaaaaaaaaaaaaaaaaaaaa`);
+		expect(wrongWords).toEqual(["This", "is", "just", "a", "really", "long", "String", "to", "test",
+		"some", "extreme", "case", "so", "the", "next", "letters", "will", "only", "be", "copies",
+		"of", "letter", "aaaaaaaaaaaaaaaaaaaaaaaaaaa"]);
+	});
+
+	it("Recognizes the words correctly", () => {
+		const result = getInvalidWords("Hallo Welt, fu. foo, bar");
+		expect(result).toEqual(["Hallo", "Welt", "foo"]);
+	});
+
+	it("Seperates words correctly along dots", () => {
+		const result = getInvalidWords("hello.world");
+		expect(result).toEqual(["hello", "world"]);
+	});
+});
+
+describe("logSingleWord", () => {
+	it("Logs a ordinary word correctly", () => {
+		const position: Position = {fromIndex: 0, toIndex: 4};
+		const issues = logSingleWord("Hello", position);
+		expect(issues).toEqual(
+			{message: `${"Hello"} von Stelle ${position.fromIndex} bis ${position.toIndex} ist ein unerlaubtes Wort! \n`,
+		position: {
+			fromIndex: position.fromIndex,
+			toIndex: position.toIndex,
+			}});
```

Comment:
Indentation. `position` should be indented once more, the closing `}` should be indented one less.